functional-tests/src/test/java/org/infinispan/online/service/endpoint/HotRodTester.java

```diff
@@ -34,20 +39,105 @@ public void testPutPerformance(URL urlToService, long timeout, TimeUnit timeUnit
       TestObjectCreator testObjectCreator = new TestObjectCreator();
 
       //when
-      while(System.currentTimeMillis() - endTime < 0) {
+      while (System.currentTimeMillis() - endTime < 0) {
          String key = testObjectCreator.getRandomString(1000);
          String value = testObjectCreator.getRandomString(1000);
          defaultCache.put(key, value);
       }
    }
 
-   private RemoteCache<String, String> getDefaultCache(URL urlToService) {
+   public RemoteCache<String, String> getDefaultCache(URL urlToService) {
+      RemoteCacheManager cachingService = getRemoteCacheManager(urlToService);
+      return cachingService.getCache();
+   }
+
+   public RemoteCache<String, String> getNamedCache(URL urlToService, String cacheName) {
+      RemoteCacheManager cachingService = getRemoteCacheManager(urlToService);
+      return cachingService.getCache(cacheName);
+   }
+
+   public RemoteCacheManager getRemoteCacheManager(URL urlToService) {
+      //given
       Configuration cachingServiceClientConfiguration = new ConfigurationBuilder()
          .addServer()
          .host(urlToService.getHost())
          .port(urlToService.getPort())
          .build();
-      RemoteCacheManager cachingService = new RemoteCacheManager(cachingServiceClientConfiguration);
-      return cachingService.getCache();
+
+      return new RemoteCacheManager(cachingServiceClientConfiguration);
+   }
+
+   public void putGetTest(URL hotRodService) {
+      RemoteCacheManager cachingService = getRemoteCacheManager(hotRodService);
+
+      stringPutGetTest(cachingService);
+      stringRemoveTest(cachingService);
+
+      intPutGetTest(cachingService);
+      byteArrayPutGetTest(cachingService);
+   }
+
+   private void byteArrayPutGetTest(RemoteCacheManager cachingService) {
+      //given
+      RemoteCache<String, byte[]> byteArrayCache = cachingService.getCache();
+      byte[] byteArrayValue = generateConstBytes(4096);
+      //when
+      byteArrayCache.put(hotRodKey, byteArrayValue);
+      //then
+      assertArrayEquals(byteArrayValue, byteArrayCache.get(hotRodKey));
+   }
+
+   private void intPutGetTest(RemoteCacheManager cachingService) {
+      //given
+      RemoteCache<String, Integer> intCache = cachingService.getCache();
+      Integer intValue = 5;
+      //when
+      intCache.put(hotRodKey, intValue);
+      //then
+      assertEquals(intValue, intCache.get(hotRodKey));
+
+   }
+
+   private void stringRemoveTest(RemoteCacheManager cachingService) {
+      //given
+      RemoteCache<String, String> stringCache = cachingService.getCache();
+      stringCache.put(hotRodKey, "value");
+      //when
+      stringCache.remove(hotRodKey);
+      //then
+      assertNull(stringCache.get(hotRodKey));
+   }
+
+   private void stringPutGetTest(RemoteCacheManager cachingService) {
+      //given
+      RemoteCache<String, String> stringCache = cachingService.getCache();
+      //when
+      stringCache.put(hotRodKey, "value");
```

Comment:
You can rewrite it into a single //when block:

```
stringCache.put(hotRodKey, "value");
String valueForCache = stringCache.get(hotRodKey);
String replacedValue = stringCache.put(hotRodKey, "newValue");
String updatedValue = stringCache.get(hotRodKey);

//then
//all asserts go here
```

Reply:
Also this tests both inserting value and updating it. My intuition tells that's two test cases actually.